tests/spec/combine_series/test_value_and_event_series.py

```diff
@@ -1,35 +1,23 @@
-from databuilder.query_language import Dataset, IdSeries, IntSeries, build_event_table
+from ..tables import e
 
-from ..helpers import transpose
+table_data = {
+    e: """
+          |  i1
+        --+-----
+        1 | 101
+        1 | 102
+        2 | 201
+        2 | 202
+    """,
+}
 
-e = build_event_table(
-    "e",
-    {
-        "patient_id": IdSeries,
-        "i": IntSeries,
-    },
-)
 
-
-def test_combine_value_and_event_series(in_memory_engine):
-    in_memory_engine.setup(
+def test_value_and_event_series(spec_test):
+    spec_test(
+        table_data,
+        (1 + e.i1).sum_for_patient(),
```

Comment:
We talked before about covering some of the cases with unit tests rather than specs. This seems like a candidate for that (along with the other inversions).

I see a couple of reasons for not including specs like this:

* When we generate docs from them they will result in what looks like duplication to a human. The repetition may make the semantics harder to grok.

* I fear we're kidding ourselves by trying to cover "all" the combinations. The only way to truly do that would be to use generative testing. (Of course this is an argument for not using unit tests for this either.)

Reply:
I was a bit torn here, but I think this is something where we probably do need to carefully exercise the full code path from QL -> QE.  Combining series of different kinds feels a bit risky to me.

> When we generate docs from them they will result in what looks like duplication to a human. The repetition may make the semantics harder to grok.

I agree this is a risk, but I think we can mitigate with good copy.

> I fear we're kidding ourselves by trying to cover "all" the combinations.

We don't have combinatorial explosion here -- we're not testing that you can combine all kinds of series with all kinds of operations, so I think this is reasonable.